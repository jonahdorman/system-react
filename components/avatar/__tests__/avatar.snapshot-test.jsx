/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import SnapshotBase from '../__examples__/base';
import SnapshotEntityIcon from '../__examples__/entity-icon';
import SnapshotEntityInitials from '../__examples__/entity-initials';
import SnapshotUserIcon from '../__examples__/user-icon';
import SnapshotUserInitials from '../__examples__/user-initials';

test('Avatar Base DOM Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotBase />
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Avatar Entity Icon Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotEntityIcon />
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Avatar Entity Initials Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotEntityInitials />
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Avatar User Icon Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotUserIcon />
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Avatar User Initials Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotUserInitials />
	).toJSON();
	expect(domTree).toMatchSnapshot();
});
