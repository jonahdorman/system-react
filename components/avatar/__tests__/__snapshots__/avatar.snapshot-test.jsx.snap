// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Base DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <img
      alt="Person Name"
      onError={[Function]}
      src="https://lightningdesignsystem.com/assets/images/avatar2.jpg"
      title="user avatar"
    />
  </span>
</div>
`;

exports[`Avatar Entity Icon Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_medium"
  >
    <span
      className="slds-icon_container slds-icon-standard-account"
      title={undefined}
    >
      <svg
        aria-hidden="true"
        className="slds-icon"
        style={undefined}
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#account"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Account icon avatar
      </span>
    </span>
  </span>
</div>
`;

exports[`Avatar Entity Initials Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_medium"
  >
    <abbr
      className="slds-avatar__initials slds-icon-standard-account"
      title="Acme Communications"
    >
      AC
    </abbr>
  </span>
</div>
`;

exports[`Avatar User Icon Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <span
      className="slds-icon_container slds-icon-standard-user"
      title={undefined}
    >
      <svg
        aria-hidden="true"
        className="slds-icon"
        style={undefined}
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#user"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        User Icon Avatar
      </span>
    </span>
  </span>
</div>
`;

exports[`Avatar User Initials Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <abbr
      className="slds-avatar__initials slds-icon-standard-user"
      title="Annie Wilson"
    >
      AW
    </abbr>
  </span>
</div>
`;
